src/extract.js

```diff
@@ -1,74 +1,54 @@
+import {setAll} from "./parents.js";
+
 /**
- * Recursively traverse the AST and return all variables and functions
+ * Recursively traverse the AST and return all top-level identifiers
  * @param {object} node  
- * @returns {object} an object with the list of variables and function
- *  in the form {variables: [...], functions: [...]}
+ * @param {object} [options]
+ * @param {function(object)} [options.filter] A function that returns true if the node should be included
+ * @param {boolean} [options.addParents] If true, add a parent property to each node
+ * @returns a list of all top-level identifiers
  */
-export default function extract(node) {
-    switch(node.type) {
```

Comment:
Nit: We use (smart) tabs, not spaces. [Read more](http://lea.verou.me/blog/2012/01/why-tabs-are-clearly-superior/)

Reply:
fixed